configure.ac

```diff
@@ -2419,6 +2419,29 @@ AC_LINK_IFELSE(
     AC_DEFINE([HAS_BSD_GETAFFINITY_NP], [1])],
     [AC_MSG_RESULT([pthread_getaffinity_np not found])])])
 
+## prctl
+AC_MSG_CHECKING([for prctl])
+AC_LINK_IFELSE(
+  [AC_LANG_SOURCE([[
+#include <sys/prctl.h>
+int main(void) {
+  return prctl(0);
+}
+  ]])],
+  [AC_DEFINE([HAS_PRCTL] , [1])
+   AC_MSG_RESULT([yes])],
+  [AC_MSG_RESULT([no])])
+
+## pthread_setname_np
+AC_CHECK_FUNCS(
+  [pthread_setname_np],
+  [AC_DEFINE([HAS_PTHREAD_SETNAME_NP], [1])])
+
+## pthread_set_name_np
+AC_CHECK_FUNCS(
+  [pthread_set_name_np],
+  [AC_DEFINE([HAS_PTHREAD_SET_NAME_NP], [1])])
+
```

Comment:
[`AC_CHECK_FUNCS([function,...])`](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Generic-Functions.html#index-AC_005fCHECK_005fFUNCS-1) defines for each _function enumerated_ in the blank-or-newline-separated argument list `HAVE_function` (in all capitals) if it is available.
I don't think you need special code to test for `prctl`, so this code could be simplified to the following, if you also switch to the `HAVE_` macros.
```suggestion
## Set thread name
AC_CHECK_FUNCS([prctl pthread_setname_np pthread_set_name_np])
```

Reply:
I added the `HAS_` definitions to follow what appears as the naming convention in the OCaml build system.

I put a special case for `prctl` because FFmpeg `configure` script seems to be testing the function presence using the same test code build as proof:

```bash
check_func_headers(){
    log check_func_headers "$@"
    headers=$1
    funcs=$2
    shift 2
    {
        for hdr in $headers; do
            print_include $hdr
        done
        echo "#include <stdint.h>"
        for func in $funcs; do
            echo "long check_$func(void) { return (long) $func; }"
        done
        echo "int main(void) { int ret = 0;"
        # LTO could optimize out the test functions without this
        for func in $funcs; do
            echo " ret |= ((intptr_t)check_$func) & 0xFFFF;"
        done
        echo "return ret; }"
    } | test_ld "cc" "$@" && enable $funcs && enable_sanitized $headers
}

check_func_headers sys/prctl.h prctl
```

I'm suspecting that this is probably because `prctl` might be declared as a `#define` in some systems which would not work with `AC_CHECK_FUNCS` which only check for the function's presence at linkage (see: https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Generic-Functions.html) which I imagine requires a proper linker symbol.

Reply:
> I added the `HAS_` definitions to follow what appears as the naming convention in the OCaml build system.

This is, I believe, purely historical.

> I'm suspecting that this is probably because `prctl` might be declared as a #define in some systems

Thanks for the explanations!
You could use [`AC_CHECK_DECLS`](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Generic-Declarations.html#index-AC_005fCHECK_005fDECLS-1)

Reply:
> > I added the `HAS_` definitions to follow what appears as the naming convention in the OCaml build system.
> 
> This is, I believe, purely historical.

Ok. I'm fine going with `HAVE_` if that's the consensus. In my experience, introducing mixed naming convention tends to create confusion.

> > I'm suspecting that this is probably because `prctl` might be declared as a #define in some systems
> 
> Thanks for the explanations! You could use [`AC_CHECK_DECLS`](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Generic-Declarations.html#index-AC_005fCHECK_005fDECLS-1)

Done!